master/buildbot/process/properties.py

```diff
@@ -80,6 +82,36 @@ def master(self):
     def master(self, value):
         self._master = value
 
+    @property
+    def sourcestamps(self):
+        if self.build is not None:
+            return self.build.getAllSourceStamps()
+        return self._sourcestamps
+
+    @sourcestamps.setter
+    def sourcestamps(self, value):
+        self._sourcestamps = value
+
+    @defer.inlineCallbacks
```

Comment:
the creator of the properties must is the setter. If it hasn't then we should raise an attribute error or something like this